src/js/components/tabs.js

```diff
@@ -0,0 +1,186 @@
+    if (!currentTab) return;
+
+    // Create an array of all the focusable elements within the modal
+    const nextTab = this.tabs.indexOf(currentTab) + 1;
+    const prevTab = this.tabs.indexOf(currentTab) - 1;
+
+    switch (event.keyCode) {
+      case keyCodes.right || keyCodes.down: {
+
+        !this.tabs[nextTab] ? this.tabs[0].focus() : this.tabs[nextTab].focus();
+
+        break;
+      }
+      case keyCodes.down: {
+
+        !this.tabs[nextTab] ? this.tabs[0].focus() : this.tabs[nextTab].focus();
+
+        break;
+      }
+      case keyCodes.left: {
+
+        !this.tabs[prevTab] ? this.tabs[this.tabs.length - 1].focus() : this.tabs[prevTab].focus();
+
+        break;
+      }
+      case keyCodes.up: {
+
+        !this.tabs[prevTab] ? this.tabs[this.tabs.length - 1].focus() : this.tabs[prevTab].focus();
+
+        break;
+      }
+      case keyCodes.end: {
+
+        this.tabs[this.tabs.length - 1].focus();
+
+        break;
+      }
+      case keyCodes.home: {
+
+        this.tabs[0].focus();
+
+        break;
+      }
+      default: {
+        break;
+      }
+    }
+  }
+
+  /**
+   * This function is used to trigger the 'tabActivated' custom event,
+   * deactivate any other tab, and activate a tab/panel pair. It removes the
+   * panel's hidden attribute, sets the tab's
+   * aria-selected attribute to true. It also allows developers to setup a
+   * custom callback function.
+   * @param {Function} callback 
+   */
+  activateTab(tab, callback) {
+    // Trigger tabActivated custom event. This event is used to control the process of switching between tabs.
+
+    const activationEvent = dispatchCustomEvent(
+      'tabActivated',
+      tab,
+      {
+        id: tab.dataset.tabPanel,
+      }
+    );
+
+    if (!activationEvent) return;
+
+    const trigger = this.element.querySelector(`[${this.tabAttribute}="${tab.dataset.tabPanel}"]`);
+
+    this.panels.forEach((panel, index) => {
+      if (panel.getAttribute(this.panelAttribute) !== tab.dataset.tabPanel && !panel.hasAttribute('hidden')) {
+        // Deactivate the appropriate tab/panel pair
+        panel.setAttribute('hidden', '');
+        this.tabs[index].setAttribute('aria-selected', 'false');
+        this.tabs[index].setAttribute('tabindex', '-1');
+      }
+    });
+
+    // Activate the appropriate tab/panel pair
+    tab.removeAttribute('hidden');
+    trigger.setAttribute('aria-selected', 'true');
+    trigger.removeAttribute('tabindex');
+
+    if (callback && typeof callback === 'function') {
+      callback();
+    }
+  }
+
+  init() {
+    this.activateTab(this.openOnInit);
+
+    this.destroy();
+
+    // Add click handlers
+    this.tablist.addEventListener('click', this._handleClick, false);
+    this.tablist.addEventListener('keydown', this._handleKeydown, false);
+  }
+
+  destroy() {
+    this.tablist.removeEventListener('click', this._handleClick, false);
+    this.tablist.removeEventListener('keydown', this._handleKeydown, false);
+  }
+
```

Comment:
Same as comment above, let's close up empty lines at the closing of nested blocks.